### PR TITLE
Add file logging to multicast receiver

### PR DESCRIPTION
**Pull request by Nickster258**
Add file logging to multicast receiver

### CODE REVIEW

File: project3/proto/pub_sub_messages.proto [hunk 1]
```diff
@@ -8,6 +8,7 @@ message Register {
   uint32 port_number = 1;
 }
 
+/// Request sent by the coordinator upon successful registration.
```
Comment (Nickster258) [inline project3/proto/pub_sub_messages.proto:11]: I have no idea how this got here. Thought I had handled all of these cases.

File: project3/participant/tasks/multicast_receiver_task.cpp [hunk 2]
```diff
@@ -46,4 +44,10 @@ thread_pool::Task::Status MulticastReceiver::RunAtomic() {
 
 void MulticastReceiver::CleanUp() { close(messenger_fd_); }
 
+void MulticastReceiver::WriteLineToLog(const std::string& message) {
+  std::ofstream log_file;
```
Comment (djpetti) [inline project3/participant/tasks/multicast_receiver_task.cpp]: Might be worth just opening the log file once in `SetUp` and keeping the handle around.

File: project3/participant/tasks/multicast_receiver_task.h [hunk 3]
```diff
@@ -31,16 +32,22 @@ class MulticastReceiver : public thread_pool::Task {
    * @param port the port this receiver will listen on
```
Comment (djpetti) [inline project3/participant/tasks/multicast_receiver_task.h:32]: Update docs

**Review by djpetti — APPROVED** [review]
Approved pending minor changes

**Pull request merged**